sharding/chainParametersHolder.go

```diff
@@ -107,29 +130,31 @@ func (c *chainParametersHolder) AllChainParameters() []config.ChainParametersByE
 	c.mutOperations.RLock()
 	defer c.mutOperations.RUnlock()
 
-	// TODO: return a copy of the object
-	return c.chainParameters
+	chainParametersCopy := make([]config.ChainParametersByEpochConfig, len(c.chainParameters))
+	for idx, chainParameterForEpoch := range c.chainParameters {
```

Comment:
use `copy(chainParametersCopy, c.chainParameters)` instead ?

Reply:
done. also added test for pointers check